sources/packages/backend/apps/api/src/institution-integration/ier-integration/ier12-file.service.ts

```diff
@@ -0,0 +1,172 @@
+    );
+    const pendingAssessments =
+      await this.studentAssessmentService.getPendingAssessment(generatedDate);
+    if (!pendingAssessments.length) {
+      return [
+        {
+          generatedFile: "none",
+          uploadedRecords: 0,
+        },
+      ];
+    }
+    this.logger.log(`Found ${pendingAssessments.length} assessments.`);
+    const fileRecords: Record<string, IER12Record[]> = {};
+    pendingAssessments.forEach((pendingAssessment) => {
+      const institutionCode =
+        pendingAssessment.offering.institutionLocation.institutionCode;
+      if (!fileRecords[institutionCode]) {
+        fileRecords[institutionCode] = [];
+      }
+      fileRecords[institutionCode].push(
+        this.createIER12Record(pendingAssessment),
+      );
+    });
+    const uploadResult: IER12UploadResult[] = [];
+    try {
+      this.logger.log("Creating IER 12 request content...");
+      for await (const ierUploadResult of Object.keys(fileRecords).map(
+        (institutionCode) =>
+          this.uploadIER12Content(institutionCode, fileRecords),
+      )) {
+        uploadResult.push(ierUploadResult);
+      }
+    } catch (error) {
+      this.logger.error(
+        `Error while uploading content for IER 12 Request: ${error}`,
+      );
+      throw error;
+    }
+    return uploadResult;
+  }
+
+  /**
+   * Upload the content in SFTP server location.
+   * @param institutionCode Institution code for the file generated.
+   * @param fileRecords Total records with institutionCode.
+   * @returns Updated records count with filepath.
+   */
+  async uploadIER12Content(
+    institutionCode: string,
+    fileRecords: Record<string, IER12Record[]>,
+  ): Promise<IER12UploadResult> {
+    try {
+      // Create the Request content for the IER 12 file by populating the content.
+      const fileContent = this.ierIntegrationService.createIER12FileContent(
+        fileRecords[institutionCode],
+      );
+      // Create the request filename with the file path for the each and every institutionCode.
+      const fileInfo = this.createRequestFileName(institutionCode);
+      this.logger.log("Uploading content...");
+      await this.ierIntegrationService.uploadContent(
+        fileContent,
+        fileInfo.filePath,
+      );
+      this.logger.log("Content uploaded.");
+      return {
+        generatedFile: fileInfo.filePath,
+        uploadedRecords: fileContent.length,
+      };
+    } catch (error) {
+      this.logger.error(
+        `Error while uploading content for IER 12 file: ${error}`,
+      );
+      throw error;
+    }
+  }
+
+  /**
+   * Create file name IER 12 records file.
+   * @param institutionCode institutionCode to be a part of the filename.
+   * @returns Full file path of the file to be saved on the SFTP.
+   */
+  createRequestFileName(institutionCode: string): {
+    fileName: string;
+    filePath: string;
+  } {
+    const timestamp = getFileNameAsCurrentTimestamp();
```

Comment:
I was under the impression that the idea was to generate a file per day in a way that the file name would be `IER_012_2022-12-16.txt`. In this way, if the same file was generated for the same day it would override the one in the SFTP.
Assuming that we need to generate 10 files and there was an error where only 8 were generated, triggering the process again would override the existing 8 and adding the other 2. Basically, the processes would be able to execute again and again for the same day not producing more files in the SFTP.

Reply:
@abschwenker just double checking that we are ignoring what I mentioned in my comment because it is not important and multiple files can be generated for the same institution and same day, right?